### PR TITLE
Catched the MessageDecodingFailedException during receive messages

### PR DESCRIPTION
**Pull request by rdotter**
Also fixes https://github.com/sroze/messenger-enqueue-transport/issues/91

After consumer is restarted, the message is rejected, so you can continue with the next message :-).

Tested with SQS and Kafka integration with Enqueue.

@weaverryan

### CODE REVIEW

**Comment by rdotter** [comment]
@weaverryan @makasim @xavismeh @shulard @Nyholm 
Can one of you check this and merge this in the upcomming release? :-)
Thanks!

**Comment by shulard** [comment]
Looks good to me !

Thank you for that PR (I haven't the merge permision but I still approve this change ^^).

File: Tests/MessageBusProcessorTest.php [hunk 1]
```diff
@@ -54,6 +55,27 @@ public function testProcess()
         $this->assertSame(Processor::ACK, $messageBusProcessor->process($message, $contextProphecy->reveal()));
     }
 
+    /**
+     * @expectedException \Symfony\Component\Messenger\Exception\MessageDecodingFailedException
```
Comment (Nyholm) [inline Tests/MessageBusProcessorTest.php]: Please dont use annotations for this. 

Use `$this->expectException()` instead. And put that line just before you expect that exception.

File: Tests/QueueInteropTransportTest.php [hunk 2]
```diff
@@ -428,6 +429,47 @@ public function testThrow()
         $transport->send($envelope);
     }
 
+    /**
+     * @expectedException \Symfony\Component\Messenger\Exception\MessageDecodingFailedException
```
Comment (Nyholm) [inline Tests/QueueInteropTransportTest.php]: Same here

File: Tests/MessageBusProcessorTest.php (hunk 1 above)
Comment (rdotter) [reply Tests/MessageBusProcessorTest.php]: @Nyholm removed the expectedException and changed into Reject message Processor (which is the correct implementation).

File: Tests/QueueInteropTransportTest.php (hunk 2 above)
Comment (rdotter) [reply Tests/QueueInteropTransportTest.php]: @Nyholm updated into expectException + added the missing reject call from the Consumer.

**Comment by rdotter** [comment]
@Nyholm can you recheck the changes (I force pushed it with the new changes).

**Comment by rdotter** [comment]
@Nyholm can you recheck and add this to another new version, we need this fix in multiple projects :-)

Thanks!!

File: QueueInteropTransport.php [hunk 3]
```diff
@@ -80,11 +81,17 @@ public function get(): iterable
...
+        try {
+            $envelope = $this->serializer->decode(array(
+                'body' => $interopMessage->getBody(),
+                'headers' => $interopMessage->getHeaders(),
+                'properties' => $interopMessage->getProperties(),
+            ));
+        } catch (MessageDecodingFailedException $e) {
+            $this->getConsumer()->reject($interopMessage);
+
+            throw $e;
```
Comment (qkdreyer) [inline QueueInteropTransport.php:93]: why are we throwing here ? it stops any `messenger:consume` running command

**Pull request merged**